modules/develop/pages/produce-data/leader-pinning.adoc

```diff
@@ -0,0 +1,45 @@
+= Leader Pinning
+:page-categories: Clients, Development
+// tag::single-source[]
+:description: Learn about leader pinning and how to configure a preferred partition leader based on availability zones.
+
+Produce requests to Redpanda topics go through the topic partition leader, which then ensure that messages are synced across its follower replicas. For a Redpanda cluster deployed across multiple availability zones (AZs), leader pinning is a mechanism to ensure that a topic's partition leaders are geographically closer to clients. Leader pinning can lower networking costs and help guarantee lower latency by routing produce requests and consume requests without xref:develop:consume-data/follower-fetching.adoc[follower fetching] to brokers located in certain AZs.
+
+== Prerequisites
+
+include::shared:partial$enterprise-license.adoc[]
+
+You must xref:manage:rack-awareness.adoc#configure-rack-awareness[configure rack awareness] for the cluster. If the config_ref:enable_rack_awareness,true,properties/cluster-properties[] cluster configuration property is set to `false`, leadership pinning is disabled across the cluster.
+
+== Configure leader pinning
+
+The following steps assume that you have Redpanda deployed in a multi-AZ setup, and you have configured each broker so that the config_ref:rack,true,properties/broker-properties[] configuration property contains rack IDs corresponding to the AZ IDs.
+
+You can set the configuration property for individual topics only, or set the cluster-wide configuration property that will enable leader pinning by default for all topics. You can also use a combination in which a default setting applies across the cluster, and you toggle the setting on or off for specific topics.
+
+* Set the topic configuration property `redpanda.leaders.preference`. The property accepts the following string values:
++
+--
+** `none` (default): Opt out the topic from leader pinning.
```

Comment:
It's not entirely accurate that `none` is the default for the topic property. Rather, the default is determined by `default_leaders_preference`.

Reply:
Topic properties in kafka can be in the "not set" state